Algorithms/Encoders/SoundexEncoder.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Linq;
+using System.Text;
+using System.Globalization;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for Soundex encoding strings.
+    /// </summary>
+    public class SoundexEncoder
+    {
+        /// <summary>
+        /// Encodes a string using the Soundex Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The Soundex encoded string (one uppercase character and three digits)</returns>
+        public string Encode(string text)
```

Comment:
Codacy found an issue: [The Cyclomatic Complexity of this method is 36 which is greater than 10 authorized.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)